### PR TITLE
chore/ch10934 Use pipeline for VRF tasks

### PR DESCRIPTION
**Pull request by connorwstein**
- Make the VRF use the pipeline 
- Added a presenter for pipeline runs which ensures that the output matches what the frontend expects in models.d.ts. Tasks which output maps, such as ethdecodeabi_log, can't be rendered in the UI because it expects strings so I marshalled them to json strings
- Since both the vrf package and the pipeline package need to make use of the low-level proving code, I pulled out a proof package which they both use to break the import cycle
- To simplify interactions with the job ORM, just made the CreateJob return a fully populated job.Job, identical to what FindJob returns

Successful UI example:
image

Error UI example:
image

### CODE REVIEW

File: core/services/job/orm.go [hunk 1]
```diff
@@ -431,27 +442,36 @@ func loadDynamicConfigVars(cfg Config, os OffchainReportingOracleSpec) *Offchain
 	}
 }
 
+func (o *orm) FindJobTx(id int32) (Job, error) {
```
Comment (connorwstein) [inline core/services/job/orm.go:445]: There are a few top level endpoints which just want to lookup a job and don't have a TxManager nor direct access to a db to create one.

**Review by samsondav — CHANGES_REQUESTED** [review]
Where is the example for the new VRF toml spec?

**Comment by connorwstein** [comment]
> Where is the example for the new VRF toml spec?

https://github.com/smartcontractkit/chainlink/pull/4729/files#diff-a39f1402fa0a50324d3c640634ee7425c82936366887cd77f7ae17cef9bcd7b5

**Review by samsondav — APPROVED** [review]

File: core/web/presenters/pipeline_run.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	// The UI expects all outputs to be strings.
+	var outputs []*string
+	if !pr.Outputs.Null {
+		outs := pr.Outputs.Val.([]interface{})
+		for _, out := range outs {
+			switch v := out.(type) {
+			case string:
+				s := v
+				outputs = append(outputs, &s)
+			case map[string]interface{}:
```
Comment (connorwstein) [inline core/web/presenters/pipeline_run.go:45]: This is the main reason for the presenter - to be able to take a map task output and convert to a string. @jkongie you may have opinions on this

**Review by samsondav — APPROVED** [review]

**Pull request merged**